src/types/web-resource.ts

```diff
@@ -0,0 +1,108 @@
+import * as TypeUtils from '../type-utils';
+import { storageRegistry } from '../storage-adapters/index';
+
+/**
+ * This is how WebResources are stored on the DB
+ */
+export type WebResourceRef = {
+	filename: string;
+	href: string;
+	contentType?: string;
+	contentDisposition?: string;
+	size?: number;
+};
+
+/**
+ * This is how WebResources are defined on creation
+ */
+export type WebResourceInput = {
+	filename: string;
+	data: Buffer;
+	contentType?: string;
+	contentDisposition?: string;
+	size?: number;
+	storage: string;
+};
+
+export const types = {
+	postgres: 'JSONB',
+	mysql: 'JSON',
+	websql: 'TEXT',
+	odata: {
+		name: 'Self.WebResource',
+		complexType: `\
+<ComplexType Name="WebResource">
+	<Property Name="filename" Nullable="false" Type="Edm.String"/>\
+	<Property Name="href" Nullable="false" Type="Edm.String"/>\
+	<Property Name="contentType" Nullable="true" Type="Edm.String"/>\
+	<Property Name="contentDisposition" Nullable="true" Type="Edm.String"/>\
+	<Property Name="size" Nullable="true" Type="Edm.Int64"/>\
+</ComplexType>`,
+	},
+};
+
+export const fetchProcessing = (data: any) => {
+	let refData: WebResourceRef;
+	if (data === null) {
+		return data;
+	}
+	if (typeof data === 'string') {
```

Comment:
Can this ever be anything other than a string?

Reply:
On PG the driver actually returns an `object` because the column is defined as `JSONB`.

On `websql` ( do we have test cases for it? ) it should return a string because the column type is defined as `TEXT`.

I'll check on MySQL, it's defined as `JSONB` there but I'm not sure what the driver will return

Reply:
Update: tried with `mysql` but the examples I have don't build. I checked the references and if using a `mysql` client, then it will return a String unless there's an explicit typecast added on the connection options.

So, for pg the client will perform the `JSON.parse` and return an `object`, and for `mysql` and `websql` we can assume a string.

Reply:
This makes sense, if nothing else then it's necessary for websql